docs/topics/wait_for.rst

```diff
@@ -0,0 +1,80 @@
+:orphan:
+.. currentmodule:: discord
+
+.. _guide_wait_for:
+
+Waiting For Events
+=============================
+When creating commands or handling events, you often find yourself wanting to wait for an action or user response. In discord.py, we use the :meth:`Client.wait_for` method. 
+
+What is ``wait_for``?
+~~~~~~~~~~~~~~~~~~~~~
+:meth:`Client.wait_for` is similar to client.event/listen, however instead of calling a function, it instead holds execution of your code until the event has been dispatched, and returns the data that was in the event. 
```

Comment:
```suggestion
:meth:`Client.wait_for` is similar to client.event/listen, however instead of calling a function, it instead halts execution of your code until the event has been dispatched, and returns the data that was in the event. 
```

Perhaps "halts" is more appropriate here?